packages/web3-types/src/eth_abi_types.ts

```diff
@@ -316,7 +316,7 @@ export type ContractMethodInputParameters<Params extends ReadonlyArray<unknown>
 			  [MatchPrimitiveType<H['type'], H['components']>, ...ContractMethodInputParameters<R>]
 			: ContractMethodInputParameters<R>
 		: Params extends undefined | unknown
-		? []
+		? any[]
```

Comment:
I'm not 100% but yours doesn't cover constructor args?

Reply:
But this should be fixed regardless of your PR

Reply:
Thanks Marin for the fix and the clarification.
It would be also great if you could add a test. As you see my PR for the methods args types did not work, so I left it for when having time for it. As I worked on it in my free time :smile: 


By the way, the constructor args types were handled as part of: https://github.com/web3/web3.js/pull/6635. Where even if the ABI was not provided, the types are inferred as possible.